source/renderer/app/stores/NetworkStatusStore.js

```diff
@@ -47,43 +46,53 @@ export default class NetworkStatusStore extends Store {
   );
   @observable _localDifficultyStartedWith = null;
 
-  _timeDifferencePollInterval: ?number = null;
-
   @action initialize() {
     super.initialize();
-    if (cachedDifficulties !== null) Object.assign(this, cachedDifficulties);
+    if (cachedState !== null) Object.assign(this, cachedState);
   }
 
-  setup() {
+  async setup() {
     this.registerReactions([
-      this._redirectToWalletAfterSync,
-      this._redirectToLoadingWhenDisconnected,
-      this._redirectToSyncingWhenLocalTimeDifferent,
-      this._pollTimeDifferenceWhenConnected,
+      this._updateSyncProgressWhenDisconnected,
+      this._updateLocalTimeDifferenceWhenConnected,
     ]);
-    this._pollSyncProgress();
+
+    // Setup polling intervals
+    this._syncProgressPollInterval = setInterval(
+      this._updateSyncProgress, SYNC_PROGRESS_INTERVAL
+    );
+    if (environment.isAdaApi()) {
+      this._updateLocalTimeDifferencePollInterval = setInterval(
```

Comment:
@DominikGuzei I fear this update is in collision with the following improvement: https://github.com/input-output-hk/daedalus/pull/719

Reply:
No it doesn't the code is even better now, you have to read through it carefully 😉